pkg/runctl/controller.go

```diff
@@ -717,7 +720,8 @@ func (c *Controller) handlePipelineRunCleaning(
 		if err = c.changeAndCommitStateAndMeter(ctx, pipelineRun, api.StateFinished, metav1.Now()); err != nil {
 			return true, err
 		}
-		if err = pipelineRun.DeleteFinalizerAndCommitIfExists(ctx); err != nil {
+		ctx, _ := extendContextLoggerWithPipelineRunInfo(origCtx, pipelineRun.GetAPIObject())
```

Comment:
Variable `logger` should not stay with the old logger, as the pair of `ctx` and `logger` would be inconsistent then. If later some code is added below that uses `logger`, it would use the old one.

```suggestion
		ctx, logger := extendContextLoggerWithPipelineRunInfo(origCtx, pipelineRun.GetAPIObject())
```